app/routes/api/auth.js

```diff
@@ -30,7 +30,7 @@ router.route('/auth/password')
 
         userPromise.then((user) => {
             user.setPassword(request.body.password, function (err, model, passwordErr) {
-                if (err || passwordErr) {
+                if (err || passwordErr || request.body.password.length < 7) {
```

Comment:
We can probably do validation in passport-local-mongoose rather than handling it manually here